lib/tic_tac_toe.ex

```diff
@@ -1,6 +1,10 @@
 defmodule TicTacToe do
-  def run do
-    IO.puts("Welcome to Tic-Tac-Toe")
-    nil
+  def run(board, display, presenter) do
+    grid = board.empty_grid
+    show_grid(grid, board, display, presenter)
+  end
+
+  def show_grid(grid, board, display, presenter) do
+    display.output(presenter.present(grid, board))
```

Comment:
Elixir has a really nice API to handle threading results through a set of functions. You could refactor this to be

```suggestion
    display.output(presenter.present(grid, board))
    grid
    |> presenter.present(@board)
    |> display.output()
```

In fact you could inline this all into run if you choose to.

```elixir 
def run(board, display, presenter) do
  board.empty_grid
  |> presenter.present(board)
  |> display.output
end

Reply:
Nice!  I'm still trying to learn the pipe operator